src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -305,10 +317,25 @@ public function setup(): void
 
         $this->exchange()->declareExchange();
 
-        $this->queue()->declareQueue();
-        $this->queue()->bind($this->exchange()->getName(), $this->queueConfiguration['routing_key'] ?? null);
+        foreach ($this->queuesConfiguration as $queueName => $queueConfig) {
+            $this->queue($queueName)->declareQueue();
+            foreach ($queueConfig['routing_keys'] ?? [] as $routingKey) {
```

Comment:
@weaverryan Just to confirm, as far as I understand, this line addresses the first point (_AMQP configuration_) I 
 made [here](https://github.com/symfony/symfony/issues/29950). Am I correct?

Reply:
I’m not sure exactly which was your first point. But this part allows you to declare that you want messenger to create multiple queues on your behalf *and* create multiple binds for each queue. I’m pretty sure this was at least one of your issues :). This part is all about creating and binding of queues - these specific lines say nothing about the publishing process or what routing keys are used.